devito/symbolics/manipulation.py

```diff
@@ -27,20 +28,31 @@ def uxreplace(expr, rule):
 
     By avoiding re-evaluations, this function is typically much quicker than
     SymPy's xreplace.
+
+    A further feature of ``uxreplace`` consists of enabling the substitution
+    of compound nodes. Consider the expression `a*b*c*d`; if one wants to replace
+    `b*c` with say `e*f`, then the following mapper may be passed:
+    `{a*b*c*d: {b: e*f, c: None}}`. This way, only the `b` and `c` pertaining to
```

Comment:
So through this way, replacing also holds if b and c do not appear in sequence...(like the one after the other) ?

Reply:
correct